agent/lib/kontena/workers/weave_worker.rb

```diff
@@ -48,19 +48,21 @@ def on_dns_add(topic, event)
     # @param [String] topic
     # @param [Docker::Event] event
     def on_container_event(topic, event)
-      if event.status == 'start'
+      if network_adapter.router_image?(event.from)
```

Comment:
Refactored as a local method call in #2207... but changed this to only do this for `start` events now, so it shouldn't matter as much.